samcli/commands/check/calculations.py

```diff
@@ -0,0 +1,115 @@
+"""
+Bottle neck calculations are done here. Warning messages are also generated here
+"""
+import click
+import boto3
+from samcli.commands._utils.resources import AWS_LAMBDA_FUNCTION
+from samcli.commands.check.resources.Graph import Graph
+from .resources.Warning import CheckWarning
+
+
+class Calculations:
+    _graph: Graph
+
+    def __init__(self, graph: Graph):
+        """
+        Args:
+            graph (Graph): The graph object. This is where all of the data is stored
+        """
+        self._graph = graph
+
+    def _generate_warning_message(
+        self,
+        capacity_used: int,
+        resource_name: str,
+        concurrent_executions: int,
+        duration: int,
+        tps: int,
+        burst_concurrency: int,
+    ):
+        """Depending on how severe the bottle neck is, a different warning message will be generated
+        Args:
+            capacity_used (int): Percentage of capacity used
+            resource_name (str): Recource name that the warning is for
+            concurrent_executions (int): Concurrent executions based on users account
+            duration (int): Duration of lambda function
+            tps (int): TPS of lambda function based on its parent resource
+            burst_concurrency (int): Burst concurrency based on users account
+        """
+        warning = CheckWarning()
+
+        if capacity_used <= 70:
```

Comment:
sorry for being OCD, but I'm curious to understand why the boundaries are set as:
```
capacity_used <= 70
70 < capacity_used < 90
90 <= capacity_used <= 100
capacity_used > 100
```
rather than:
```
capacity_used < 70
70 <= capacity_used < 90
90 <= capacity_used < 100
capacity_used >= 100
```
or 
```
capacity_used <= 70
70 < capacity_used <= 90
90 < capacity_used <= 100
capacity_used > 100
```

Nit: you can omit the lower boundaries in the `if...elif...else` statements. For example,
```
if capacity_used < 70:
    ...
elif capacity_used < 90:
    ...
elif capacity_used < 100:
    ...
else:
    ...
```

Reply:
1. Stefano and I discussed it and these are the boundaries he wanted. I can double check with him if he still wants these.
2. There is no `elif capacity_used > 100` in the code if that's what you're referring to. There is a comment stating what the `else` is. I like to see what the else is for without having to look through the entire `if else` statement.

Reply:
1. I'm good with the numbers of the boundaries (i.e. 70, 90, 100). I think my question is - why are the comparison signs not consistent in all cases? (i.e. either `<` on the right and `<=` on the left, or `<=` on the right and `<` on the left)
2. It's fine to use just `else` to represent the case of `capacity_used > 100`, assuming `capacity_used` is a number. And the comment is good too 👍 and I think you should still keep it. My example for omitting the lower boundary does the exact same thing as yours, but I personally think my example looks cleaner.

Reply:
1. Stefano defined where each warning should be placed depending on the percentage of concurrency used. Making the signs consistent in either of your examples would not match what Stefano wants. 
2. I understand what you're saying now. I will change it.